pandas/tools/plotting.py

```diff
@@ -886,10 +891,11 @@ def _iter_data(self, data=None, keep_index=False, fillna=None):
 
         from pandas.core.frame import DataFrame
         if isinstance(data, (Series, np.ndarray, Index)):
+            label = self.label if self.label else data.name
```

Comment:
Need to be careful here. We need to treat `False` differently than `None`, in case someone wants the actual label in the legend to be the text "False". do `label = self.label if self.label is not None else data.name`.